src/main/java/net/schmizz/sshj/sftp/RandomAccessRemoteFile.java

```diff
@@ -318,10 +318,8 @@ public void writeShort(int v)
     public void writeUTF(String str)
             throws IOException {
         final DataOutputStream dos = new DataOutputStream(rf.new RemoteFileOutputStream(fp));
-        try {
+        try (dos) {
```

Comment:
This changes is not supported on Java 8 according to the automated tests:

```
/home/runner/work/sshj/sshj/src/main/java/net/schmizz/sshj/sftp/RandomAccessRemoteFile.java:321: error: variables in try-with-resources are not supported in -source 8
        try (dos) {
                ^
  (use -source 9 or higher to enable variables in try-with-resources)
  ```
  
It looks like `fp` could be updated inside the try block.